examples/material-ui-remix-ts/app/root.tsx

```diff
@@ -1,128 +1,43 @@
-import * as React from 'react';
 import {
   Links,
-  LiveReload,
   Meta,
   Outlet,
   Scripts,
   ScrollRestoration,
-  useRouteError,
-  isRouteErrorResponse,
-} from '@remix-run/react';
-import { withEmotionCache } from '@emotion/react';
-import { unstable_useEnhancedEffect as useEnhancedEffect } from '@mui/material';
-import theme from './src/theme';
-import ClientStyleContext from './src/ClientStyleContext';
-import Layout from './src/Layout';
+} from "@remix-run/react";
 
-interface DocumentProps {
-  children: React.ReactNode;
-  title?: string;
-}
-
-const Document = withEmotionCache(({ children, title }: DocumentProps, emotionCache) => {
-  const clientStyleData = React.useContext(ClientStyleContext);
+import { getMuiLinks } from "./mui/getMuiLinks";
+import { MuiMeta } from "./mui/MuiMeta";
+import { LinksFunction } from "@remix-run/node";
+import { MuiDocument } from "./mui/MuiDocument";
 
-  // Only executed on client
-  useEnhancedEffect(() => {
-    // re-link sheet container
-    emotionCache.sheet.container = document.head;
-    // re-inject tags
-    const tags = emotionCache.sheet.tags;
-    emotionCache.sheet.flush();
-    tags.forEach((tag) => {
-      // eslint-disable-next-line no-underscore-dangle
-      (emotionCache.sheet as any)._insertTag(tag);
-    });
-    // reset cache to reapply global styles
-    clientStyleData.reset();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+export const links: LinksFunction = () => [...getMuiLinks()];
 
+export function Layout({ children }: { children: React.ReactNode }) {
   return (
     <html lang="en">
       <head>
         <meta charSet="utf-8" />
-        <meta name="viewport" content="width=device-width,initial-scale=1" />
-        <meta name="theme-color" content={theme.palette.primary.main} />
-        {title ? <title>{title}</title> : null}
+        <meta name="viewport" content="width=device-width, initial-scale=1" />
         <Meta />
+        <MuiMeta />
         <Links />
-        <link rel="preconnect" href="https://fonts.googleapis.com" />
-        <link rel="preconnect" href="https://fonts.gstatic.com" crossOrigin="" />
-        <link
-          rel="stylesheet"
-          href="https://fonts.googleapis.com/css2?family=Roboto:wght@300;400;500;700&display=swap"
-        />
-        <meta name="emotion-insertion-point" content="emotion-insertion-point" />
       </head>
       <body>
         {children}
         <ScrollRestoration />
         <Scripts />
-        <LiveReload />
       </body>
     </html>
   );
-});
+}
 
-// https://remix.run/docs/en/main/route/component
-// https://remix.run/docs/en/main/file-conventions/routes
 export default function App() {
   return (
-    <Document>
-      <Layout>
+    <>
+      <MuiDocument>
```

Comment:
Why not adding `<CssBaseline />` here instead of the extra `<MuiDocument>`?

Reply:
Same as I explained above: Simple components for single responsibility.